src/libs/antares/series/series.cpp

```diff
@@ -61,6 +61,47 @@ static std::string errorMessage(const std::map<std::string, const TimeSeries*>&
     return msg.str();
 }
 
+uint TimeSeriesNumbers::height() const {
+    return tsNumbers.height;
+}
+
+uint32_t TimeSeriesNumbers::operator[](uint y) const
+{
+    return tsNumbers[0][y];
+}
+
+uint32_t& TimeSeriesNumbers::operator[](uint y)
+{
+    return tsNumbers[0][y];
+}
+
+void TimeSeriesNumbers::reset(uint h)
+{
+    tsNumbers.reset(1, h);
+}
+
+namespace // anonymous
+{
+struct TSNumbersPredicate
+{
+    uint32_t operator()(uint32_t value) const
+    {
+        return value + 1;
+    }
+};
+} // namespace
+
+void TimeSeriesNumbers::clear()
+{
+    tsNumbers.clear();
+}
+
+void TimeSeriesNumbers::saveToBuffer(std::string& data) const
+{
+    static TSNumbersPredicate predicate;
```

Comment:
We could probably use a lambda here and remove the struct, the function will be more self-contained.
Maybe it can also be `constexpr`?
By the way predicate seems like an odd name but it's an issue of the matric class !

Reply:
Done ! https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2085/commits/50067f8e167197b9d409ac4c1782b0061f8cc9d5

It seems to work based on the fact that expression `object()` is valid both for `TSNumbersPredicate` (with `operator()`) and for a lambda function.